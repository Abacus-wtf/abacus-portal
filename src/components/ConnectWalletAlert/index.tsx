import React, { FunctionComponent } from "react"
import { Alert } from "shards-react"
import styled from "styled-components"
import Button from "@components/Button"
import { useToggleWalletModal } from "@state/application/hooks"

const FlexContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`

const AlertCopy = styled.div`
  padding-bottom: 10px;
`

export const CONNECT_WALLET_BUTTON_ID = "CONNECT_WALLET_BUTTON_ID"

const ConnectWalletAlert: FunctionComponent = () => {
  const toggleWalletModal = useToggleWalletModal()
  return (
    <Alert theme="light" style={{ backgroundColor: "white" }}>
      <FlexContainer style={{ alignItems: "center" }}>
        <AlertCopy>
          You cannot participate until you connect your wallet
        </AlertCopy>
        <Button
          data-testid={CONNECT_WALLET_BUTTON_ID}
          onClick={() => toggleWalletModal()}
        >
          Connect Wallet
        </Button>
      </FlexContainer>
    </Alert>
  )
}

export default ConnectWalletAlert
