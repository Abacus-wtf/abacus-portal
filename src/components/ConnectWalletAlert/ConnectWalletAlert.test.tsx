import { render } from "@test-utils/index"
import React from "react"
import "@testing-library/jest-dom/extend-expect"
import ConnectWalletAlert from "."

const defaultProps = {}

describe("ConnectWalletAlert", () => {
  it("Matches snapshot", () => {
    const component = render(<ConnectWalletAlert {...defaultProps} />)
    expect(component.container).toMatchSnapshot()
  })
})
