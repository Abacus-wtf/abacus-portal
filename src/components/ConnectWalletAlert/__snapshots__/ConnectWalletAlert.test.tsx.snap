// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectWalletAlert Matches snapshot 1`] = `
<div>
  <div
    class="alert alert-light fade"
    role="alert"
    style="background-color: white;"
  >
    <div
      class="sc-hKgJUU looSwH"
      style="align-items: center;"
    >
      <div
        class="sc-eCstlR dnXMhV"
      >
        You cannot participate until you connect your wallet
      </div>
      <button
        class="sc-bdfBQB jjXNlU btn btn-primary"
        type="button"
      >
        Connect Wallet
      </button>
    </div>
  </div>
</div>
`;
